src/dataset-builder/CohortEditor.ts

```diff
@@ -52,17 +52,28 @@ export const CriteriaView = (props: CriteriaViewProps) => {
 
   const [criteriaCount, setCriteriaCount] = useLoadedData<DatasetParticipantCountResponse>();
 
-  useEffect(() => {
-    setCriteriaCount(async () =>
-      DataRepo()
-        .dataset(datasetId)
-        .getCounts({
-          // Create a "cohort" to get the count of participants for this criteria on its own.
-          cohorts: [{ criteriaGroups: [{ criteria: [criteria], name: '', meetAll: true, mustMeet: true }], name: '' }],
-        })
-    );
-  }, [criteria, datasetId, setCriteriaCount]);
+  const debounceSetCriteriaCount = useRef(
```

Comment:
The name doesn't seem right to me. This is really debouncing the call to `getCounts()`; the fact that it also debounces setting the value once that API returns seems incidental, as debouncing the set operation has no effect (vs debouncing the API call, which does).

Maybe a better name is `updateCriteriaCount` which describes what this function is doing instead of how it's doing it?

Reply:
I agree that we aren't really debouncing setCriteriaCount but more calling debouncing `getCounts();` and then setting it.